### PR TITLE
Fixed ms test runnner

### PR DESCRIPTION
**Pull request by andyfenna**

### CODE REVIEW

**Comment by andyfenna** [comment]
the test runner must be called with the is_ms_test parameter set... see below:-

        desc "Run unit tests"
        test_runner :sqltest do |mstest|
            mstest.files = ['C:\dbgit\db\build\SqlTests.dll']
            mstest.is_ms_test
            mstest.exe = "C:/Program Files (x86)/Microsoft Visual Studio 11.0/Common7/IDE/MSTest.exe"
            mstest.native_exe
        end

**Comment by haf** [comment]
Excellent =)

Write a blurb in the wiki that `is_ms_test` exists, and I'll merge this.

**Comment by andyfenna** [comment]
updated wiki to correspond to changes

File: lib/albacore/task_types/nugets_pack.rb [hunk 1]
```diff
@@ -193,12 +197,17 @@ def nuget_gem_exe
       def opts
         files = @files.respond_to?(:each) ? @files : [@files]
 
-        [:authors, :description, :version].each do |required|
-          warn "metadata##{required} is missing from nugets_pack [nugets pack: config]" if @package.metadata.send(required) == 'MISSING' 
-        end
+          if (@nuspec  == '')
```
Comment (haf) [inline lib/albacore/task_types/nugets_pack.rb]: So what if it's `nil`? Doesn't it initialize to nil by default?

Comment (haf) [inline lib/albacore/task_types/nugets_pack.rb]: Is the extra two space indentation intended?

**Comment by haf** [comment]
Put in a few comments. There's also the `unless` keyword that checks for the inverse of `if`. If `@nuspec` is nil by default then you could do `unless @nuspec ; ... normal path ... ; else ... `

File: lib/albacore/task_types/nugets_pack.rb (hunk 1 above)
Comment (andyfenna) [inline lib/albacore/task_types/nugets_pack.rb]: Your're right, I'll make the changes as mentioned above ;)

**Comment by andyfenna** [comment]
maybe your ci test needs changing to refect these changes. how do i go about changing these?

**Comment by haf** [comment]
No the test is correct...

You just haven't changed all of your checks `== ''`. Like this one: https://github.com/andyfenna/albacore/blob/master/lib/albacore/task_types/nugets_pack.rb#L248

View the diff of this PR to see all the places you have edited. If you can split this into two separate PRs; one for the msbuild flag which was correct then I can help you correct your other nuspec code separately.

**Comment by andyfenna** [comment]
Thanks for the help. It's much appreciated? I'll check through things later. On a curve ball does your gem PUSH? To a source?

Sent from my iPhone

On 5 May 2016, at 21:07, Henrik Feldt <notifications@github.com<mailto:notifications@github.com>> wrote:


No the test is correct...

You just haven't changed all of your checks == ''. Like this one: https://github.com/andyfenna/albacore/blob/master/lib/albacore/task_types/nugets_pack.rb#L248

View the diff of this PR to see all the places you have edited. If you can split this into two separate PRs; one for the msbuild flag which was correct then I can help you correct your other nuspec code separately.

-

**Comment by haf** [comment]
Yes, the `release` task does

**Comment by andyfenna** [comment]
i've resolved that earlier problem, however the CI has still failed - unsure why.

I'm calling the task like this...

    desc 'package nugets - finds all projects and package them'
    nugets_pack :pack => [:versioning] do |p|
        p.nuspec   = 'nuget/app.nuspec'
        p.out     = "#{configatron.dir.build}"
        p.exe     = "#{configatron.dir.rake_tools}/nuget-v2/nuget.exe"
    end

and the console returns the below..

C:\dbgit\App>rake nuget:pack
C:\dbgit\App\rake-tools\nuget-v2\NuGet.exe Pack -OutputDirectory C:/dbgit/App/build C:/dbgit/App/nuget/App.nuspec
Attempting to build package from 'App.nuspec'.
Successfully created package 'C:/dbgit/App/build\App.0.1.0.0.nupkg'.
I 2016-05-06T10:44:04.272779+01:00: symbols not configured for generation, use Config#gen_symbols to do so [nugets pack: cmd]

So i think we need a new CI test as we are passing in p.nuspec attribute.

**Comment by andyfenna** [comment]
All tests are now passing. I'll update wiki now with new feature for nugests_pack - can this be merged once done?

File: lib/albacore/task_types/nugets_pack.rb [hunk 2]
```diff
@@ -193,12 +193,15 @@ def nuget_gem_exe
       def opts
         files = @files.respond_to?(:each) ? @files : [@files]
 
-        [:authors, :description, :version].each do |required|
-          warn "metadata##{required} is missing from nugets_pack [nugets pack: config]" if @package.metadata.send(required) == 'MISSING' 
+        unless !@nuspec.nil?
```
Comment (haf) [inline lib/albacore/task_types/nugets_pack.rb]: Why not just `if @nuspec`?

**Comment by andyfenna** [comment]
The UNLESS statement gets your head in a spin sometimes lol, all code changes have been made as above.

I have looked at the release task, how would I call it from a rake file like the nugets_pack task?

At the minute I'm using it like this... i'm sure its correct though...

namespace :nuget do
   Albacore::Tasks::Release.new :release,
        pkg_dir: "#{configatron.dir.build}",
        nuget_exe: "#{configatron.dir.rake_tools}/nuget-v2/nuget.exe",
        nuget_source: "http://nuget.source.com",
        api_key: 'xxxxxxx'
end

This is throwing this error when trying to execute. 

"rake nuget:release"

"Don't know how to build task 'nuget:release'"

Any ideas?

**Comment by haf** [comment]
I think that the release task doesn't take its surrounding namespace. You should be able to call it with `:release` plainly.

**Comment by haf** [comment]
@andyfenna Thank you!

**Comment by andyfenna** [comment]
When will this be available for release? v2.5.8 does not include this change

**Comment by haf** [comment]
v2.5.9 released

**Pull request merged**